docs/extensibility.md

```diff
@@ -27,19 +30,118 @@ To create an Upgrade Assistant extension, you will need to start with a manifest
+
+## Analyzers/code fixers
+Roslyn analyzers and code fix providers. Upgrade Assistant's source updater step looks in the dependency injection container for any analyzers with associated code fix providers and will include them in the sub-steps to the source updater step. So, by registering their own Roslyn analyzers and code fix providers, extenders can customize the source update steps used by Upgrade Assistant.
+
+## Updaters
+Various services may request an implementation of `IUpdater<TUpdater>` which provides a way to update an object of type `TUpdater`. An example is a `IUpdater<ConfigFile>` that will provide udpates for configuration files (`app.config`, `web.config`).
+
+This generic parameter can be extended by any other service to provide a custom way of defining an updater.
+
+## Dependency analyzers
+Dependency analysis can be extended in multiple ways, including code and simple configuration.
+
+Some built-in dependency analyzers perform the following actions:
+  1. Removes packages that other referenced packages depend on (transitive dependencies). PackageReference-style references only need to include top level packages so this analyzer removes package references that appear to be transitive so that only top-level dependencies are included in the csproj.
+  2. If a referenced NuGet package isn't compatible with the target .NET version but a newer version of the NuGet package is, the package updater step automatically updates the version to the first major version that will work.
+  3. If a `PackageMap` is defined, NuGet references will be replaced based on specific packages listed in configuration files. For example, there's a config setting that specifically indicates `System.Threading.Tasks.Dataflow` should replace `TPL.Dataflow`.
+
+Package maps are one of the non-code ways to extend the dependency analysis step. An extension's `PackageMapPath` path can contain json files that map old packages to new ones. In each set, there are old (.NET Framework) package references which should be removed and new (.NET 5.0/Standard) package references which should be added. If a project being upgraded by the tool contains references to any of the 'NetFrameworkPackages' references from a set, those references will be removed and all of the 'NetCorePackages' references from that same set will be added. If old (NetFrameworkPackage) package references specify a version, then the references will only be replaced if the referenced version is less than or equal to that version.
+
+By adding package map files, users can supply customized rules about which NuGet package references should be removed from upgraded projects and which NuGet package references (if any) should replace them.
```

Comment:
May want to mention that this can also be used to add or remove Framework references (not just package references).